From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Thu, 15 May 2026 00:00:00 -0400
Subject: [PATCH] Complete absl/log targets in system absl BUILD

TF 2.21.0 / XLA reference more @com_google_absl//absl/log targets than
the systemlib BUILD declared (vlog_is_on, globals, initialize, flags,
log_entry, log_sink, log_sink_registry, log_streamer, absl_check,
scoped_mock_log). Declare them all, mapped to the conda-forge libabseil
shared objects.
---
diff --git a/third_party/xla/third_party/absl/system.absl.log.BUILD b/third_party/xla/third_party/absl/system.absl.log.BUILD
index 141bb319..87760cc8 100644
--- a/third_party/xla/third_party/absl/system.absl.log.BUILD
+++ b/third_party/xla/third_party/absl/system.absl.log.BUILD
@@ -3,19 +3,13 @@ load("@rules_cc//cc:defs.bzl", "cc_library")
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
-    name = "die_if_null",
-    linkopts = ["-labsl_die_if_null"],
-    deps = [
-        ":log",
-        "//absl/base:config",
-        "//absl/base:core_headers",
-        "//absl/strings",
-    ],
+    name = "vlog_is_on",
+    linkopts = ["-labsl_vlog_config_internal"],
 )
 
 alias(
-    name = "absl_log",
-    actual = ":log",
+    name = "absl_vlog_is_on",
+    actual = ":vlog_is_on",
 )
 
 cc_library(
@@ -27,6 +21,17 @@ cc_library(
         "-labsl_log_internal_message",
         "-labsl_log_internal_nullguard",
     ],
+    deps = [":vlog_is_on"],
+)
+
+alias(
+    name = "absl_log",
+    actual = ":log",
+)
+
+cc_library(
+    name = "log_streamer",
+    deps = [":log"],
 )
 
 cc_library(
@@ -38,3 +43,75 @@ cc_library(
         "-labsl_log_internal_nullguard",
     ],
 )
+
+alias(
+    name = "absl_check",
+    actual = ":check",
+)
+
+cc_library(
+    name = "die_if_null",
+    linkopts = ["-labsl_die_if_null"],
+    deps = [
+        ":log",
+        "//absl/base:config",
+        "//absl/base:core_headers",
+        "//absl/strings",
+    ],
+)
+
+cc_library(
+    name = "globals",
+    linkopts = [
+        "-labsl_log_globals",
+        "-labsl_log_severity",
+        "-labsl_log_internal_globals",
+        "-labsl_vlog_config_internal",
+        "-labsl_raw_logging_internal",
+    ],
+)
+
+cc_library(
+    name = "initialize",
+    linkopts = ["-labsl_log_initialize"],
+    deps = [":globals"],
+)
+
+cc_library(
+    name = "flags",
+    linkopts = [
+        "-labsl_log_flags",
+        "-labsl_vlog_config_internal",
+    ],
+    deps = [":globals"],
+)
+
+cc_library(
+    name = "log_entry",
+    linkopts = [
+        "-labsl_log_entry",
+        "-labsl_log_internal_proto",
+    ],
+)
+
+cc_library(
+    name = "log_sink",
+    linkopts = ["-labsl_log_sink"],
+    deps = [":log_entry"],
+)
+
+cc_library(
+    name = "log_sink_registry",
+    linkopts = ["-labsl_log_internal_log_sink_set"],
+    deps = [":log_sink"],
+)
+
+cc_library(
+    name = "scoped_mock_log",
+    testonly = 1,
+    deps = [
+        ":log_entry",
+        ":log_sink",
+        ":log_sink_registry",
+    ],
+)
